### ISSUE TITLE
Many simultaneous jobs may cause connectivity issue with the API.

### ISSUE THREAD

**Issue by willgraf** [vanvalenlab/kiosk-benchmarking#43]
When benchmarking many jobs (10,000+) the `Job` clients can get stuck in a fail loop with a variety of error messages:

- No route to host. (NOT a server issue).
- Connection Timeout
- TaskStopped (seems like just noise from `twisted`, nothing to worry about)
- Connection reset by peer. (likely due to the server being at [too high a load](https://www.quora.com/What-does-the-104-Connection-reset-by-peer-error-notification-mean-and-how-do-I-fix-it).)
-  502 Bad Gateway (likely due to a request being sent to a server that just came up).

A simple fix for this could be to reduce the `UPDATE_INTERVAL` from its default of 15 seconds to something less frequent like 60 seconds.  A more general fix may include an exponential backoff when a request has failed.

**Comment by willgraf** [comment]
I believe the `No route to host` error will be resolved by #44, which prevents all outgoing ports from getting full of connections by including a "Connection: close" header in each request.

**Comment by willgraf** [comment]
It looks like there is definitely work to do in preventing some performance-related errors on the server. The most common error logs from `Job._retry_post_request`:

```
Encountered ResponseNeverReceived during REDIS HGET status: [<twisted.python.failure.Failure twisted.internet.error.ConnectionLost: Connection to the other side was lost in a non-clean fashion.>]

Encountered RequestTransmissionFailed during REDIS HGET status: [<twisted.python.failure.Failure twisted.internet.error.ConnectionDone: Connection was closed cleanly.>]

Encountered ConnectError during REDIS HGET status: An error occurred while connecting: 104: Connection reset by peer.
```

All of the above errors seem to be server-related rather than client related.

Server size issues may also cause a number of other errors mentioned above, including:
- `502 Bad Gateway`
- `Connection Timeout`
